src/main/java/io/jlu/alfredo/Alfredo.java

```diff
@@ -18,37 +18,83 @@
 
 public class Alfredo extends ListenerAdapter {
 
-    private final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
-    private static Map<String, Command> commandMap = new HashMap<>();
+    private static final String CREDENTIALS_FILE = "credentials.txt";
+    private static final String TOKEN_FILE = "token.txt";
+    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
     public static void main(String[] args) throws LoginException, IOException {
+        getInstance().init(CREDENTIALS_FILE, TOKEN_FILE);
+    }
 
-        File file = new File("credentials.txt");
-        BufferedReader br = new BufferedReader(new FileReader(file));
+    /**
+     * Singleton instance
+     */
+    private static Alfredo INSTANCE = null;
 
-        String jdbcUrl = br.readLine();
-        Jdbi jdbi = Jdbi.create(jdbcUrl, br.readLine(), br.readLine());
+    private Map<String, Command> commandMap = new HashMap<>();
 
-        JDABuilder builder = new JDABuilder(AccountType.BOT);
-        file = new File("token.txt");
-        br = new BufferedReader(new FileReader(file));
+    private Alfredo() {}
 
-        String token = br.readLine();
+    public static Alfredo getInstance() {
```

Comment:
Dubious use of singleton pattern - may as well just make the Alfredo constructor public since you're only using this for tests